src/index.ts

```diff
@@ -144,4 +146,32 @@ export function snakeCase<T extends string | readonly string[]>(str?: T) {
   return kebabCase(str || "", "_") as SnakeCase<T>;
 }
 
+export function flatCase(): "";
+export function flatCase<T extends string | readonly string[]>(
+  str: T,
+): FlatCase<T>;
+export function flatCase<T extends string | readonly string[]>(str?: T) {
+  return kebabCase(str || "", "") as FlatCase<T>;
+}
+
+export function trainCase(): "";
+export function trainCase<T extends string | readonly string[]>(
+  str: T,
+): TrainCase<T>;
+export function trainCase<T extends string | readonly string[]>(str?: T) {
+  return str
+    ? ((
+        (Array.isArray(str) ? str : splitByCase(str as string)) as Array<string>
+      )
+        // eslint-disable-next-line unicorn/no-array-reduce -- map and filter would require iterating twice
+        .reduce((acc, p) => {
+          acc +=
+            acc.length > 0
+              ? p && `-${upperFirst(p.toLowerCase())}`
+              : p && upperFirst(p.toLowerCase());
```

Comment:
Although it is all fine to lowecase the 2nd+ chars, I'm wondering if we might do it similar to other utils when `{ normalize: true }` flag is provided. Particularly, `WWW-Authenticate` will become `Www-Authenticate` this wat I suppose which is not desired at least for HTTP headers.

Reply:
It is actually very dependent on the implementation. PHP in some versions hardly enforces HTTP Header Case (Content-type won’t work, but Content-Type does), although RFC 2616 says that header names are case insensitive.
WWW- headers are an interesting case and one of the exceptions. For most headers, MDN and especially browsers are using Train-Case as a convention.
In HTTP/2 there is an ambiguous definition that says headers are insensitive, but must be converted to kebap-cass.
I would therefore keep the name Train-Case as it is the most commonly found one when searching for it on the web and is not strictly bound to HTTP headers, although mostly used there.

Reply:
Adding normalize is a great idea to handle the edge cases. This would also handle headers like `Accept-CH` or headers containing `XSS/CSRF`.

Reply:
I added the normalize option to the function, types and tests.